test/unittests/t_util.cc

```diff
@@ -32,7 +38,94 @@ class ThreadDummy {
 };
 
 
-TEST(T_Util, ThreadProxy) {
+class T_Util : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    const bool retval = MkdirDeep(string(sandbox), 0700);
+    ASSERT_TRUE(retval) << "failed to create string(sandbox)";
+  }
+
+  virtual void TearDown() {
+    const bool retval = RemoveTree(string(sandbox));
+    ASSERT_TRUE(retval) << "failed to remove string(sandbox)";
+  }
+
+ protected:
+  static void WriteBuffer(int fd) {
+    sleep(1);
```

Comment:
Yes, it could. I was trying not to use functions that are going to be tested in order to avoid bug creations, but with this one there shouldn't be any problem.